zephyr/lib/alloc.c

```diff
@@ -341,6 +377,7 @@ void *rballoc_align(uint32_t flags, uint32_t caps, size_t bytes,
 /*
  * Free's memory allocated by above alloc calls.
  */
+__attribute__((optimize("-O0")))
```

Comment:
Best to say why this is needed with an inline comment, It could be tool/platform specific.

Reply:
indeed... why is this needed at all?

Reply:
fixed